homework_5/pd_and_visualization.py

```diff
@@ -0,0 +1,420 @@
+# importing the libraries needed for this task
+
+import pandas as pd
+import numpy as np
+import re # i need it for the task 2
+import matplotlib.pyplot as plt
+import seaborn as sns
+from matplotlib import rc
+from matplotlib.ticker import (MultipleLocator, AutoMinorLocator)
+
+
+# task 1: create two functions for reading .gff and .bed formats
+
+def read_gff(path_to_gff):
+    """ function to read the .gff file """
+    column_names=["chromosome","source","type","start","end","score","strand","phase","attributes"] # set the column names
+    gff_data = pd.read_csv(path_to_gff, sep='\t', names=column_names) # read the file
+    gff_data = gff_data[gff_data.chromosome.str.startswith("#") == False] # skip the commented lines
```

Comment:
С фильтрацией комментов после прочтения норм вариант. Есть правда более простой, с небольшим нюансом, но тем не менее очень широко используемый
```suggestion
    gff_data = pd.read_csv(path_to_gff, sep='\t', names=column_names, comment="#") # read the file
```